### PR TITLE
chore: Remove declareLegacyNamespace() from files in core

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm test`.

### Proposed Changes
Removes declareLegacyNamespace() from all remaining files under core except msg.js, which has special object creation logic that complicates things.

### CODE REVIEW

File: core/blockly.js [hunk 1]
```diff
@@ -430,11 +492,85 @@ exports.Block = Block;
...
+exports.browserEvents = browserEvents;
+exports.bumpObjects = bumpObjects;
+exports.clipboard = clipboard;
+exports.common = common;
+exports.connectionTypes = connectionTypes;
+exports.constants = constants;
+exports.dialog = dialog;
+exports.inject = inject;
+exports.inputTypes = inputTypes;
+exports.internalConstants = internalConstants;
```
Comment (rachel-fenichel) [inline core/blockly.js:573]: Don't export this one--these are all aliased in this file to their old names, but I don't want to make the new (internal) names visible.

File: core/flyout_metrics_manager.js [hunk 2]
```diff
@@ -11,7 +11,6 @@
 'use strict';
 
 goog.module('Blockly.FlyoutMetricsManager');
```
Comment (rachel-fenichel) [inline core/flyout_metrics_manager.js:13]: Missing a re-export.

**Review by rachel-fenichel — APPROVED** [review]
Two nits then lgtm

File: core/blockly.js (hunk 1 above)
Comment (gonfunko) [reply core/blockly.js]: Done

File: core/flyout_metrics_manager.js (hunk 2 above)
Comment (gonfunko) [reply core/flyout_metrics_manager.js:13]: Done

**Pull request merged**